agent/command.go

```diff
@@ -64,16 +62,17 @@ func (a *Agent) runCommands(ctx context.Context, tc *taskContext, commands []mod
 				tc.taskConfig.Expansions.Put(key, newVal)
 			}
 
+			tc.setCurrentTimeout(a.getTimeout(commandInfo))
```

Comment:
Timeout should apply only to non-post/non-timeout tasks. These should be governed by defaultCallbackCmdTimeout.